lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvStatement.php

```diff
@@ -264,6 +264,9 @@ private function prepare()
                     SQLSRV_PHPTYPE_STREAM(SQLSRV_ENC_BINARY),
                     SQLSRV_SQLTYPE_VARBINARY('max'),
                 ];
+            } elseif (PDO::PARAM_STR === $this->types[$column]) {
+                $variable = (string) $variable;
```

Comment:
This cast should not be applied. This makes any type mismatch failures silent.

A `StringCastableObject` should be mapped via a custom DBAL type that allows string castable objects instead.